lightspark-sdk/src/commonMain/kotlin/com/lightspark/sdk/LightsparkCoroutinesClient.kt

```diff
@@ -205,6 +206,44 @@ class LightsparkCoroutinesClient private constructor(
         )
     }
 
+    /**
+     * Creates a Lightning invoice for the given node. This should only be used for generating invoices for LNURLs, with
+     * [LightsparkCoroutinesClient.createInvoice] preferred in the general case.
+     *
+     * @param nodeId The ID of the node for which to create the invoice.
+     * @param amountMsats The amount of the invoice in milli-satoshis.
+     * @param metadata The LNURL metadata payload field from the initial payreq response. This will be hashed and
+     * present in the h-tag (SHA256 purpose of payment) of the resulting Bolt 11 invoice.
+     */
+    suspend fun createLnurlInvoice(
+        nodeId: String,
+        amountMsats: Long,
+        metadata: String,
+    ): InvoiceData {
+        requireValidAuth()
+
+        val md = MessageDigest.getInstance("SHA-256")
```

Comment:
I can throw this in a separate function under `utils` if you'd like as well @jklein24 .